### PR TITLE
Update span_selector.py

### PR DESCRIPTION
**Pull request by Jashshah6086**
## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by Jashshah6086** [comment]
Updated span_selector and all checks passed

**Comment by jklymak** [comment]
Can you explain a bit more in the PR description what this does and why?  Was this example broken before?

**Comment by Jashshah6086** [comment]
The example was not robust enough.
If the cursor was not drag it gave error in command line.
This was because the list was empty. So, added condition on the length of list(region_x) to prevent any error.

**Comment by QuLogic** [comment]
This is related to #20200; please give your PRs a summary as not everyone is going to remember and cross-reference every single issue.

**Comment by Jashshah6086** [comment]
Okay, this is my first time contributing so didn't know.

File: examples/widgets/span_selector.py [hunk 1]
```diff
@@ -32,10 +32,12 @@ def onselect(xmin, xmax):
...
-    line2.set_data(region_x, region_y)
-    ax2.set_xlim(region_x[0], region_x[-1])
-    ax2.set_ylim(region_y.min(), region_y.max())
-    fig.canvas.draw()
+
+    if len(region_x) >= 2:
+        line2.set_data(region_x, region_y)
+        ax2.set_xlim(region_x[0], region_x[-1])
+        ax2.set_ylim(region_y.min(), region_y.max())
+        fig.canvas.draw()
```
Comment (timhoffm) [inline examples/widgets/span_selector.py:40]: I'd rather use 
```
if xmin == xmax:
    return
```
at the top of the function, which is a bit clearer on the semantics.

**Review by timhoffm — COMMENTED** [review]
Tip for future pull requests: If you write "Closes #20200" this will link the PR to the issue and also automatically close the issue when the PR gets merged. See also [Linking a pull request to an issue using a keyword](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: examples/widgets/span_selector.py (hunk 1 above)
Comment (Jashshah6086) [reply examples/widgets/span_selector.py:40]: This is not true, because there is possibility of empty list which will give error in this case.

Comment (timhoffm) [reply examples/widgets/span_selector.py:40]: You're right.

**Review by timhoffm — APPROVED** [review]

**Comment by timhoffm** [comment]
Thanks @Jashshah6086! Congratulations on your first contribution to Matplotlib. We hope to see you again.

**Pull request merged**